tests/wrapper/test_pagamentos.py

```diff
@@ -73,97 +79,152 @@ def test_cancelar_pagamento_1(self):
         """
         Teste para verificar a URL da requisição e dados
         """
-        PagamentoLoteBBWrapper().cancelar_pagamentos("1", "2", "3", "x")
-
-        url = PagamentoLoteBBWrapper()._construct_url("cancelar-pagamentos")
-
+        request_url = PagamentoLoteBBWrapper()._construct_url("cancelar-pagamentos")
         expected_json = {
             "agenciaDebito": "2",
             "contaCorrenteDebito": "3",
             "digitoVerificadorContaCorrente": "x",
             "listaPagamentos": [{"codigoPagamento": "1"}],
         }
-
-        self.mocked_post.assert_called_with(
-            url, **self.get_request_complements(), json=expected_json
+        self.mock_responses.add(
+            responses.POST,
+            request_url,
+            headers=self._build_authorization_header(1),
+            json=expected_json,
```

Comment:
o que acontece se não for adicionado a resposta com a url correta?

E o que acontece se for requisitado duas vezes :thinking:

Reply:
> o que acontece se não for adicionado a resposta com a url correta?

Vai lançar um erro dizendo que a requisição não está registrada no mock e falha o teste já que a consulta com a URL, mesmo que errada, não foi requisitada.

Por exemplo, se cadastrarmos somente as requisições de autenticação e a consulta de um lote de pagamentos. Ao tentarmos cadastrar um lote de pagamentos, a autenticação vai ocorrer normalmente. Mas na segunda requisição, o cadastro do lote de pagamentos, uma exceção vai ser lançada porque não foi registrada no mock e falha porque a consulta não foi chamada.


> E o que acontece se for requisitado duas vezes thinking

As requisições ocorrem normalmente, mas aí tem que ajustar os asserts de acordo.